packages/core/logging/core-logging-server-internal/src/logger_adapter.test.ts

```diff
@@ -88,3 +88,129 @@ test('forwards all method calls to new internal logger if it is updated.', () =>
   expect(newInternalLogger.trace).toHaveBeenCalledTimes(1);
   expect(newInternalLogger.trace).toHaveBeenCalledWith('trace-message', undefined);
 });
+
+describe('global meta', () => {
+  let internalLogger: Logger;
+
+  beforeEach(() => {
+    internalLogger = {
+      debug: jest.fn(),
+      error: jest.fn(),
+      fatal: jest.fn(),
+      info: jest.fn(),
+      log: jest.fn(),
+      trace: jest.fn(),
+      warn: jest.fn(),
+      get: jest.fn(),
+    };
+  });
+
+  afterEach(() => {
+    Object.values(internalLogger).forEach((val) => (val as jest.Mock<() => void>).mockReset());
+  });
+
+  ['trace', 'debug', 'info', 'warn', 'error', 'fatal'].forEach((method) => {
+    test(`inserts global meta in ${method} entries`, () => {
+      const adapter = new LoggerAdapter(internalLogger, new Map([['a.b.c', `${method}: d`]]));
```

Comment:
This should be addressed in my refactoring. The logic to merge the global context in with each log entry doesn't handle deconfliction since the object is already flattened at this point, so I added a test to the logging system instead.